src/Services/Authentication/Authentication.BusinessLogic/Services/Implementations/UserService.cs

```diff
@@ -0,0 +1,91 @@
+using Authentication.BusinessLogic.DTOs;
+using Authentication.BusinessLogic.DTOs.Request;
+using Authentication.BusinessLogic.DTOs.Response;
+using Authentication.BusinessLogic.Exceptions;
+using Authentication.BusinessLogic.Providers.Interfaces;
+using Authentication.BusinessLogic.Services.Interfaces;
+using Authentication.DataLayer.Models;
+using Authentication.DataLayer.Repositories.Interfaces;
+using AutoMapper;
+using Shared.Constants;
+using Shared.Models;
+
+namespace Authentication.BusinessLogic.Services.Implementations;
+
+public class UserService(IUserRepository _userRepository, IMapper _mapper) : IUserService
+{
+    public async Task<GeneralResponseDto> DeleteUserByIdAsync(string userId, CancellationToken cancellationToken)
+    {
+        var user = await _userRepository.GetByIdAsync(userId, cancellationToken);
+        
+        if (user is null)
+        {
+            throw new NotFoundException(userId);
+        }
+
+        var result = await _userRepository.DeleteUserByIdAsync(user);
+        
+        if (!result.Succeeded)
+        {
+            throw new DeleteUserException(ExceptionMessages.DeleteUserFailed);
+        }
+        
+        return new GeneralResponseDto() { Message = "User deleted successfully" };
+    }
+    
+    public async Task<GeneralResponseDto> UpdateUser(UserRequestDto userDto)
```

Comment:
Always add the suffix "Async" to the names of methods that have an Async or async modifie